src/community/ogcapi/ogcapi-core/src/main/java/org/geoserver/ogcapi/QueryablesBuilder.java

```diff
@@ -55,15 +56,21 @@ public QueryablesBuilder forType(FeatureTypeInfo ft) throws IOException {
         this.queryables.setCollectionId(ft.prefixedName());
         this.queryables.setTitle(Optional.ofNullable(ft.getTitle()).orElse(ft.prefixedName()));
         this.queryables.setDescription(ft.getDescription());
-        return forType((SimpleFeatureType) ft.getFeatureType());
+        return forType(ft.getFeatureType());
     }
 
-    public QueryablesBuilder forType(SimpleFeatureType ft) {
+    public QueryablesBuilder forType(FeatureType ft) {
         Map<String, Schema> properties =
-                ft.getAttributeDescriptors().stream()
+                ft.getDescriptors().stream()
+                        .filter(
+                                ad -> // ignore feature chaining links, they might be duplicated
+                                !ad.getName()
+                                                .equals(
+                                                        ComplexFeatureConstants
+                                                                .FEATURE_CHAINING_LINK_NAME))
```

Comment:
Ugly indenting, need to work a bit along with the formatter. For example, a static import on the FEATURE_CHAINING_LINK_NAME, and/or moving the inline comment before ``ad ->``

Reply:
Fixed formatting issue